File: app/src/main/java/org/metabrainz/mobile/presentation/features/links/LinksViewModel.java
package org.metabrainz.mobile.presentation.features.links;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import org.metabrainz.mobile.data.sources.api.entities.Link;

import java.util.List;

public class LinksViewModel extends ViewModel {

    private final MutableLiveData<List<Link>> linksLiveData = new MutableLiveData<>();

    public LinksViewModel() {
    }

    LiveData<List<Link>> getData() {
        return linksLiveData;
    }

    public void setData(List<Link> links) {
        linksLiveData.setValue(links);
    }
}

File: app/src/main/java/org/metabrainz/mobile/presentation/features/links/LinksViewModel.kt
package org.metabrainz.mobile.presentation.features.links

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import org.metabrainz.mobile.data.sources.api.entities.Link

class LinksViewModel : ViewModel() {
    private val linksLiveData = MutableLiveData<List<Link>>()
    val data: LiveData<List<Link>>
        get() = linksLiveData

    fun setData(links: List<Link>) {
        linksLiveData.value = links
    }
}